Settings UI: Related Posts controls and preview + indentation for controls after toggle

eliorivero opened the pull request:
Fixes #6037

#### Changes proposed in this Pull Request:

## Indentation
- All settings that are child of a toggle will be indented, as well as explanation related to a toggle immediately above.
- They're wrapped with `div.jp-form-has-child` that doesn't add margin on its own but delegates to the first `.jp-form-fieldset` and `.jp-form-setting-explanation` inside it.

Related conversation in Slack p1485210196000238-slack-jetpack-settings-ui

This is how Related Posts look like this with the indentation and the updated styling
<img width="732" alt="captura de pantalla 2017-01-24 a las 16 44 12" src="https://cloud.githubusercontent.com/assets/1041600/22263781/60eb225e-e254-11e6-9123-a18cad7e6096.png">

## Related Posts
- add the traditional controls to manage headline and thumbnails
- toggles for headline and thumbnail work instantly and the preview is updated instantly
![related](https://cloud.githubusercontent.com/assets/1041600/21656171/ceedab64-d29b-11e6-80d7-368bada82e25.gif)

#### Testing instructions:
* go to Jetpack > Traffic > Related posts and click on the toggles.

cc @rickybanister @MichaelArestad

rickybanister commented:
Awesome. A couple of thingsI noticed: 

* Once the main toggle is turned on, the sub-toggles should be indented (I think it's 24px, @MichaelArestad can confirm)
* Can we put the subtext about the customizer above the preview, rather than above the sub-toggles? That way it will be more like `Here's a preview your related posts. If you'd like to modify these settings while looking at your site you can _use the customizer_.`
* Some margins between things need some finessing.

Looks great!

MichaelArestad commented:
The child settings (subtoggles) should all be indented 24px (width of toggle) + 12px (width of label margin) for a total of 36px. We might update the label indent to 16px to fit our grid better in which case, 40px would be ideal. For now, I'd go with 36px.

MichaelArestad commented:
Needs the same styling fixes as everything else in the new Settings UI and could use the indent in this PR for the child settings. Other than the child indent, this PR looks good.

zinigor commented:
Code-wise it's all great, and works well, thank you!

rickybanister commented:
Are the visuals up-to-date in this PR?

The current visuals reveal a couple of problems. 

* We've lost any kind of 'title' on the preview. 
* The italic description about using the customizer is still in an awkward place. As per my comment from three weeks ago, can we incorporate that text into a title for the preview?

Let's discuss this later today @MichaelArestad

eliorivero commented:
@rickybanister I've updated the styling of Related Posts, included `Preview`, moved the customizer link, and updated the static screenshot that displays the indentation and styling.

rickybanister commented:
This got merged before a final review. We are still not consistent with the Calypso interface. @MichaelArestad let's talk about the question's I asked over here: https://github.com/Automattic/wp-calypso/pull/10925

On WordPress.com we are showing the Preview when the setting is toggled off.
In Jetpack we are hiding the Preview until the setting is toggled on. 

Let's pick a path and make it the same for both.
